File: C2_numberInEnglish/py/numberInEnglish.test.py
import unittest
from numberInEnglish import number_in_english
 
class Test(unittest.TestCase):

    def test_input_type(self):
        self.assertRaises(Exception, number_in_english, "145")
        self.assertRaises(Exception, number_in_english, "num")
        self.assertRaises(Exception, number_in_english, True)
        self.assertRaises(Exception, number_in_english, None)
        self.assertRaises(Exception, number_in_english, 100.95)
        self.assertRaises(Exception, number_in_english, [100.95])
        self.assertRaises(Exception, number_in_english, (10, 11, 12))

    def test_single_digits(self):
 
        self.assertEqual("one", number_in_english(1))
        self.assertEqual("two", number_in_english(2))
        self.assertEqual("three", number_in_english(3))
        self.assertEqual("four", number_in_english(4))
        self.assertEqual("five", number_in_english(5))
        self.assertEqual("six", number_in_english(6))
        self.assertEqual("seven", number_in_english(7))
        self.assertEqual("eight", number_in_english(8))
        self.assertEqual("nine", number_in_english(9))

    def test_teens(self):
 
        self.assertEqual("eleven", number_in_english(11))
        self.assertEqual("twelve", number_in_english(12))
        self.assertEqual("thirteen", number_in_english(13))
        self.assertEqual("fourteen", number_in_english(14))
        self.assertEqual("fifteen", number_in_english(15))
        self.assertEqual("sixteen", number_in_english(16))
        self.assertEqual("seventeen", number_in_english(17))
        self.assertEqual("eighteen", number_in_english(18))
        self.assertEqual("nineteen", number_in_english(19))

    def test_tens(self):
 
        self.assertEqual("ten", number_in_english(10))
        self.assertEqual("twenty", number_in_english(20))
        self.assertEqual("thirty", number_in_english(30))
        self.assertEqual("forty", number_in_english(40))
        self.assertEqual("fifty", number_in_english(50))
        self.assertEqual("sixty", number_in_english(60))
        self.assertEqual("seventy", number_in_english(70))
        self.assertEqual("eighty", number_in_english(80))
        self.assertEqual("ninety", number_in_english(90))

    def test_hundreds(self):
 
        self.assertEqual("one hundred", number_in_english(100))
        self.assertEqual("two hundred", number_in_english(200))
        self.assertEqual("three hundred", number_in_english(300))
        self.assertEqual("four hundred", number_in_english(400))
        self.assertEqual("five hundred", number_in_english(500))
        self.assertEqual("six hundred", number_in_english(600))
        self.assertEqual("seven hundred", number_in_english(700))
        self.assertEqual("eight hundred", number_in_english(800))
        self.assertEqual("nine hundred", number_in_english(900))
        self.assertEqual("one hundred twenty five", number_in_english(125))
        self.assertEqual("four hundred sixty eight", number_in_english(468))
        self.assertEqual("nine hundred ten", number_in_english(910))

    def test_thousands(self):
 
        self.assertEqual("one thousand", number_in_english(1000))
        self.assertEqual("two thousand", number_in_english(2000))
        self.assertEqual("three thousand", number_in_english(3000))
        self.assertEqual("four thousand", number_in_english(4000))
        self.assertEqual("five thousand", number_in_english(5000))
        self.assertEqual("six thousand", number_in_english(6000))
        self.assertEqual("seven thousand", number_in_english(7000))
        self.assertEqual("eight thousand", number_in_english(8000))
        self.assertEqual("nine thousand", number_in_english(9000))
        self.assertEqual("two thousand one hundred twenty five", number_in_english(2125))
        self.assertEqual("nine thousand four hundred sixty eight", number_in_english(9468))
        self.assertEqual("three thousand nine hundred ten", number_in_english(3910))
        self.assertEqual("seventy six thousand three hundred ninety two", number_in_english(76392))
        self.assertEqual("five hundred five thousand seven hundred twenty", number_in_english(505720))

    def test_large_numbers(self):
 
        self.assertEqual("nine hundred ninety nine thousand nine hundred ninety eight", number_in_english(999998))
        self.assertEqual("twelve million forty three", number_in_english(12000043))
        self.assertEqual("ten million five hundred twenty six thousand six hundred thirty three", number_in_english(10526633))
        self.assertEqual("three million five hundred sixty thousand seventy eight", number_in_english(3560078))
        self.assertEqual("one billion", number_in_english(1000000000))
        self.assertEqual("nine billion nine hundred eighty three million one hundred thousand two hundred twenty five", number_in_english(9983100225))
        self.assertEqual("nine billion nine hundred ninety nine million nine hundred ninety nine thousand nine hundred ninety nine", number_in_english(9999999999))        
 
if __name__ == '__main__':
  unittest.main()

File: C2_numberInEnglish/js/numberInEnglish.test.js
var numberInEnglish = require('./numberInEnglish.js');
var should = require('should');

describe('numberInEnglish', function() {

  it('should be a function', function() {
    numberInEnglish.should.be.a.Function;
  });

  it('should write single digits', function () {
    numberInEnglish(0).should.equal('zero');
    numberInEnglish(1).should.equal('one');
    numberInEnglish(2).should.equal('two');
    numberInEnglish(-3).should.equal('negative three');
    numberInEnglish(4).should.equal('four');
    numberInEnglish(-5).should.equal('negative five');
    numberInEnglish(6).should.equal('six');
    numberInEnglish(-7).should.equal('negative seven');
    numberInEnglish(8).should.equal('eight');
    numberInEnglish(-9).should.equal('negative nine');
  });

  it('should write teens', function () {
    numberInEnglish(11).should.equal('eleven');
    numberInEnglish(-12).should.equal('negative twelve');
    numberInEnglish(13).should.equal('thirteen');
    numberInEnglish(14).should.equal('fourteen');
    numberInEnglish(-15).should.equal('negative fifteen');
    numberInEnglish(16).should.equal('sixteen');
    numberInEnglish(17).should.equal('seventeen');
    numberInEnglish(18).should.equal('eighteen');
    numberInEnglish(-19).should.equal('negative nineteen');
  });

  it('should write tens', function () {
    numberInEnglish(10).should.equal('ten');
    numberInEnglish(20).should.equal('twenty');
    numberInEnglish(-30).should.equal('negative thirty');
    numberInEnglish(40).should.equal('forty');
    numberInEnglish(50).should.equal('fifty');
    numberInEnglish(60).should.equal('sixty');
    numberInEnglish(-70).should.equal('negative seventy');
    numberInEnglish(80).should.equal('eighty');
    numberInEnglish(-90).should.equal('negative ninety');
    numberInEnglish(44).should.equal('forty four');
    numberInEnglish(-67).should.equal('negative sixty seven');
    numberInEnglish(99).should.equal('ninety nine');
  });

  it('should write hundreds', function () {
    numberInEnglish(-100).should.equal('negative one hundred');
    numberInEnglish(200).should.equal('two hundred');
    numberInEnglish(300).should.equal('three hundred');
    numberInEnglish(400).should.equal('four hundred');
    numberInEnglish(-500).should.equal('negative five hundred');
    numberInEnglish(600).should.equal('six hundred');
    numberInEnglish(700).should.equal('seven hundred');
    numberInEnglish(800).should.equal('eight hundred');
    numberInEnglish(-900).should.equal('negative nine hundred');
    numberInEnglish(125).should.equal('one hundred twenty five');
    numberInEnglish(-468).should.equal('negative four hundred sixty eight');
    numberInEnglish(910).should.equal('nine hundred ten');
  });

  it('should write thousands', function () {
    numberInEnglish(-1000).should.equal('negative one thousand');
    numberInEnglish(2000).should.equal('two thousand');
    numberInEnglish(-3000).should.equal('negative three thousand');
    numberInEnglish(4000).should.equal('four thousand');
    numberInEnglish(5000).should.equal('five thousand');
    numberInEnglish(-6000).should.equal('negative six thousand');
    numberInEnglish(7000).should.equal('seven thousand');
    numberInEnglish(-8000).should.equal('negative eight thousand');
    numberInEnglish(9000).should.equal('nine thousand');
    numberInEnglish(50000).should.equal('fifty thousand');
    numberInEnglish(-2125).should.equal('negative two thousand one hundred twenty five');
    numberInEnglish(9468).should.equal('nine thousand four hundred sixty eight');
    numberInEnglish(-76392).should.equal('negative seventy six thousand three hundred ninety two');
    numberInEnglish(505720).should.equal('five hundred five thousand seven hundred twenty');
  });

  it('should write large numbers', function () {
    numberInEnglish(1000000).should.equal('one million');
    numberInEnglish(-1986655).should.equal('negative one million nine hundred eighty six thousand six hundred fifty five');
    numberInEnglish(3560078).should.equal('three million five hundred sixty thousand seventy eight'); 
    numberInEnglish(-2385024).should.equal('negative two million three hundred eighty five thousand twenty four');
    numberInEnglish(12000043).should.equal('twelve million forty three');
    numberInEnglish(-10526633).should.equal('negative ten million five hundred twenty six thousand six hundred thirty three');
    numberInEnglish(999999999).should.equal('nine hundred ninety nine million nine hundred ninety nine thousand nine hundred ninety nine');
    numberInEnglish(-998310022).should.equal('negative nine hundred ninety eight million three hundred ten thousand twenty two');
    numberInEnglish(1000000000).should.equal('one billion');
    numberInEnglish(-1000000000).should.equal('negative one billion');
  });

});
